deploy/vagrant/bin/init.py

```diff
@@ -13,10 +15,19 @@
 import sys
 from sys import stderr
 import argparse
+import json
+import tempfile
+import errno
+
+import boto
+from boto import ec2
+import yaml
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument('provider')
+    parser.add_argument('--is-aws-spot', action='store_true')
```

Comment:
I think it is a little bit messy that the script that sets everything up, also has a mode that just returns true or false.

Also, is this script only for 'aws'? What happens if the provider is not 'aws' but is-aws-spot or prepare-aws-spot is used?